### PR TITLE
Issue #13693: migrate few metric modules to property macro

### PR DESCRIPTION
**Pull request by romani**
Issue #13693

setMax: https://github.com/checkstyle/checkstyle/commit/c044038c3b932586c0d384b614254cc2d1838838#diff-d554a793e3a3cbe81e95a9dd9b699444150c22c39a4c35ee7696fbdd04805d14R109 ( Jan 2, 2004)
https://sourceforge.net/projects/checkstyle/files/OldFiles/
image

setExcludedClass: (Dec 2, 2013) , it is 5.7
https://github.com/checkstyle/checkstyle/commit/cd7d040b4d2c4a80e87ef9b5be9bbd964f19e97d#diff-d554a793e3a3cbe81e95a9dd9b699444150c22c39a4c35ee7696fbdd04805d14R109
dates are weird a bit for few releases of 5.X:
image

setExcludeClassesRegexps:
https://github.com/checkstyle/checkstyle/commit/7b95603c0e5036f397365a05b68483d0da690c63#diff-ae3adc6efccc0a0cca1c89bf35cb69fd68ae8d0ce40c52176814218bd4e144a7R133
tags are weird ..... it isnot 4.4
https://github.com/checkstyle/checkstyle/issues/3234, it is 7.7

setExcludedPackages:
https://github.com/checkstyle/checkstyle/commit/57785f3aa7d71b67a3c8adc8b60a008f921c33c8#diff-ae3adc6efccc0a0cca1c89bf35cb69fd68ae8d0ce40c52176814218bd4e144a7R152
tags are weird
it is 7.7 by issue - https://github.com/checkstyle/checkstyle/issues/3309

### CODE REVIEW

**Comment by romani** [comment]
Semaphore failure is not related

**Comment by romani** [comment]
@rnveach , CI is green , I think all is ready for review.

**Comment by romani** [comment]
all since tags are reverted, map is used.

**Comment by romani** [comment]
version auto discovery is postponed .
to let us focus on migration to macros (property wording sync).

@rnveach , please let me know, if you want to skip such PR and be back in  loop for updates for versions.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java [hunk 1]
```diff
@@ -48,7 +48,8 @@
  * <ul>
  * <li>
  * Property {@code excludeClassesRegexps} - Specify user-configured regular
- * expressions to ignore classes.
+ * expressions to ignore classes. All excluded packages should end with a period,
+ * so it also appends a dot to a package name.
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java:52]: Where did `All excluded packages should end with a period, so it also appends a dot to a package name.` come from?

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java]: removed.

**Comment by romani** [comment]
@nrmancuso , PR is rebased

**Comment by romani** [comment]
rebased one more time

**Review by nrmancuso — APPROVED** [review]

**Pull request merged**